src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        threshold = (int) LOAD_FACTOR * DEFAULT_CAPACITY;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        validatedIndex(indexOf(key));
+        if (size == threshold) {
+            resize();
+        }
+        Node<K, V> node = new Node<>(key, value, null);
+        if (table[indexOf(key)] == null) {
+            table[indexOf(key)] = node;
+            size++;
+        } else {
+            Node<K, V> currentNode = table[indexOf(key)];
+            while (currentNode != null) {
+                if (currentNode.key == key || currentNode.key != null
+                        && currentNode.key.equals(key)) {
+                    currentNode.value = value;
+                    break;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = node;
+                    size++;
+                    break;
+                }
+                currentNode = currentNode.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        validatedIndex(indexOf(key));
+        Node<K, V> node = table[indexOf(key)];
+        while (node != null) {
+            if (node.key == key || node.key != null && node.key.equals(key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int indexOf(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private void validatedIndex(int index) {
+        if (index < 0 || index >= table.length) {
+            throw new ArrayIndexOutOfBoundsException("Index" + index + "is invalid");
+        }
+    }
```

Comment:
Can you get index < 0 or >= table/length when call indexOf() method? Do you need this method?